### PR TITLE
Remove unnecessary/unused elements from (C)KF

### PR DESCRIPTION
**Pull request by FabianKlimpel**
This PR removes unnecessary template parameters from the the (C)KF and the gain matrix formulation. Additionally unused variables, typedefs and a function are removed in this cleanup.

### CODE REVIEW

File: Core/include/Acts/Fitter/GainMatrixSmoother.hpp [hunk 1]
```diff
@@ -48,15 +47,15 @@ class GainMatrixSmoother {
...
+  Result<void> operator()(const GeometryContext& gctx,
+                          MultiTrajectory<source_link_t>& trajectory,
+                          size_t entryIndex) const {
     ACTS_VERBOSE("Invoked GainMatrixSmoother on entry index: " << entryIndex);
     using namespace boost::adaptors;
 
     // using ParVector_t = typename parameters_t::ParVector_t;
-    using CovMatrix = typename parameters_t::CovMatrix_t;
-    using GainMatrix = CovMatrix;
+    using Matrix_t =
```
Comment (XiaocongAi) [inline Core/include/Acts/Fitter/GainMatrixSmoother.hpp:57]: Following the naming convention, shall we just use `Matrix`?

**Comment by XiaocongAi** [comment]
Hi @FabianKlimpel , this PR looks good to me except that I'm not so sure of the removal of input/output converter. Maybe @paulgessinger or @asalzburger could comment on this.  On my side, I will approve.

File: Core/include/Acts/Fitter/GainMatrixSmoother.hpp (hunk 1 above)
Comment (FabianKlimpel) [reply Core/include/Acts/Fitter/GainMatrixSmoother.hpp]: Changed

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 2]
```diff
@@ -1037,7 +971,7 @@ class KalmanFitter {
     }
 
     // Return the converted Track
-    return m_outputConverter(std::move(kalmanResult));
+    return std::move(kalmanResult);
```
Comment (paulgessinger) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:974]: The move here is not needed I think.

File: Core/include/Acts/Fitter/KalmanFitter.hpp [hunk 3]
```diff
@@ -1143,8 +1069,8 @@ class KalmanFitter {
...
```
Comment (paulgessinger) [inline Core/include/Acts/Fitter/KalmanFitter.hpp:1072]: Same here.

File: Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp [hunk 4]
```diff
@@ -1227,9 +1176,9 @@ class CombinatorialKalmanFilter {
     }
 
     // Return the converted Track
-    return m_outputConverter(std::move(combKalmanResult));
+    return std::move(combKalmanResult);
```
Comment (paulgessinger) [inline Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp:1179]: And this one.

**Comment by msmk0** [comment]
@paulgessinger My apologies for marking the conversations as resolved. I though these were old ones.

File: Core/include/Acts/Fitter/KalmanFitter.hpp (hunk 2 above)
Comment (FabianKlimpel) [reply Core/include/Acts/Fitter/KalmanFitter.hpp]: done

File: Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp (hunk 4 above)
Comment (FabianKlimpel) [reply Core/include/Acts/TrackFinder/CombinatorialKalmanFilter.hpp]: done

File: Core/include/Acts/Fitter/KalmanFitter.hpp (hunk 3 above)
Comment (FabianKlimpel) [reply Core/include/Acts/Fitter/KalmanFitter.hpp]: done

**Pull request merged**